saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/servlet/filter/Saml2WebSsoAuthenticationFilter.java

```diff
@@ -83,6 +88,7 @@ public Saml2WebSsoAuthenticationFilter(AuthenticationConverter authenticationCon
 		this.authenticationConverter = authenticationConverter;
 		setAllowSessionCreation(true);
 		setSessionAuthenticationStrategy(new ChangeSessionIdAuthenticationStrategy());
+		setAuthenticationRequestRepository(new HttpSessionSaml2AuthenticationRequestRepository());
```

Comment:
I realize that this is what was done with `setSessionAuthenticationStrategy`; however, it's preferable to not call overrideable methods in a constructor. It may be better to do:

```java
private Saml2AuthenticationRequestRepository authenticationRequestRepository =
    new HttpSessionAuthenticationRequestRepository();
```